lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -919,11 +983,9 @@ SILBasicBlock *SingleExitLoopTransformer::createNewExitBlockWithDemux(
     return newBlock;
   };
 
-  // Create a new exit block.
-  // FIXME: We can avoid creating an additional block and instead connect the
-  // header directly to the demuxBlock created in the loop below. Alternatively,
-  // we can also use contractUncondBranches in TFParititon.cpp to remove this
-  // block later.
+  // Create a new exit block. Strictly, we don't need this block, but it makes
```

Comment:
"we don't need this block" or "we don't always need this block"? If the former, what does "if appropriate" 2 lines below mean?